cmd/cp-main.go

```diff
@@ -288,7 +288,7 @@ func doCopyFake(cpURLs URLs, pg Progress) URLs {
 }
 
 // doPrepareCopyURLs scans the source URL and prepares a list of objects for copying.
-func doPrepareCopyURLs(ctx context.Context, session *sessionV8, cancelCopy context.CancelFunc) (totalBytes, totalObjects int64) {
+func doPrepareCopyURLs(ctx context.Context, session *sessionV8, cancelCopy context.CancelFunc) (totalBytes, totalObjects int64, errSeen bool) {
```

Comment:
Consider returning a struct. Callers are getting rather ugly.

Reply:
That was my original though, but I felt like it would be too much of a refactor. I'll get it done.

Reply:
I decided to leave this one alone. But the other functions were all refactored